bin/pyntlm_auth/app.py

```diff
@@ -0,0 +1,148 @@
+import os
+import binascii
+
+from flask import Flask, g, request, jsonify
+
+from samba import param
+from samba.dcerpc import netlogon, ntlmssp, srvsvc
+from samba.dcerpc.netlogon import (netr_Authenticator, netr_WorkstationInformation, MSV1_0_ALLOW_MSVCHAPV2)
+from samba.credentials import Credentials, DONT_USE_KERBEROS
+from samba.dcerpc.misc import SEC_CHAN_WKSTA, SEC_CHAN_DOMAIN, SEC_CHAN_BDC
+
+from configparser import ConfigParser
+
+conf_path = os.getenv("CONF")
+listen_port = os.getenv("LISTEN")
+
+config = ConfigParser()
+try:
+    with open(conf_path, 'r') as file:
+        config.read_file(file)
+
+    if 'AD' in config:
+        netbios_name = config.get('AD', 'netbios_name')
+        realm = config.get('AD', 'realm')
+        server_string = config.get('AD', 'server_string')
+        workgroup = config.get('AD', 'workgroup')
+        server_name = config.get('AD', 'server_name')  # we need a valid DNS server or,  adds a DNS record in hosts file
+        workstation = config.get('AD', 'workstation')
+        username = config.get('AD', 'username')
+        password = config.get('AD', 'password')
+        password_is_nt_hash = config.get('AD', 'password_is_nt_hash')
+        domain = config.get('AD', 'domain')
+    else:
+        print("The specified section does not exist in the config file.")
+except FileNotFoundError:
+    print("The specified config file does not exist.")
+
+except configparser.Error as e:
+    print(f"Error reading config file: {e}")
+
+app = Flask(__name__)
+
+machineCred = None
+secureChannelConn = None
+
+
+def initGlobalSecureConnection():
+    global machineCred
+    global secureChannelConn
+
+    lp = param.LoadParm()
+
+    try:
+        lp.load("/root/default.conf")
+    except KeyError:
+        raise KeyError("SMB_CONF_PATH not set")
+
+    lp.set('netbios name', netbios_name)
+    lp.set('realm', realm)
+    lp.set('server string', server_string)
+    lp.set('workgroup', workgroup)
+
+    machineCred = Credentials()
+
+    machineCred.guess(lp)
+    machineCred.set_secure_channel_type(SEC_CHAN_WKSTA)
+    machineCred.set_kerberos_state(DONT_USE_KERBEROS)
+
+    machineCred.set_workstation(workstation)
+    machineCred.set_username(username)
+    machineCred.set_password(password)
+
+    machineCred.set_password_will_be_nt_hash(True if password_is_nt_hash == "1" else False)
+    machineCred.set_domain(domain)
+
+    secureChannelConn = netlogon.netlogon("ncacn_np:%s[schannel,seal]" % server_name, lp, machineCred)
```

Comment:
You should add proper error handling here. Like:

1. invalid username
2. invalid password
3. account expired/locked

Try with different values/use-cases. We want to give *very good insights* on why things could possibly go wrong.